### PR TITLE
Animal Magic: Added test case for GenerateWandEnergy()

### PR DESCRIPTION
**Pull request by funwithbots**
Proposed test case to improve testing of GenerateWandEnergy() task of Animal Magic.
The tests currently pass if a student uses randomly generated ints and then typecasts the result to float64 which doesn't meet the requirements.

`tests` was bumped to ensure that even if a random seed is used, the test should pass. For grins, I ran the bucket test 1M times with various values for `tests` and it passed at 150. I bumped to 200 for extra margin.

Assuming no issues, I will give the same treatment to the other two tasks in the exercise.
#2417

### CODE REVIEW

File: exercises/concept/animal-magic/animal_magic_test.go [hunk 1]
```diff
@@ -55,11 +58,22 @@ func TestWandEnergy(t *testing.T) {
...
 	}
+
+	var emptyBuckets int
+	for _, v := range buckets {
+		if v == 0 {
+			emptyBuckets++
+		}
+	}
+	if emptyBuckets > 1 {
+		t.Errorf("GenerateWandEnergy() results are not uniformly distributed. %d of %d buckets are empty.", emptyBuckets, numBuckets)
```
Comment (junedev) [inline exercises/concept/animal-magic/animal_magic_test.go:75]: For concept exercises, the student cannot see the test file when working on the website. That means it is important that the error messages are understandable without seeing the code.

I would recommend not talk about "buckets", instead either write a general statement that some sections of the range are not covered by the results or and output one example of an empty section (which you would need to "remember" from the loop above).

**Review by junedev — COMMENTED** [review]
Thanks for working on this. I had just one comment regarding the failure message.

File: exercises/concept/animal-magic/animal_magic_test.go (hunk 1 above)
Comment (funwithbots) [reply exercises/concept/animal-magic/animal_magic_test.go:75]: Makes sense. Also no reason to run the entire loop once a failure is detected.

**Review by junedev — APPROVED** [review]

**Comment by junedev** [comment]
I reduced the precision for low and high because there were quite a lot of zeros at end. Worked great otherwise. Will merge this in a bit, thanks again.

**Pull request merged**